executable-spec/src/Cardano/Ledger/Spec/STS/Update/Hupdate.hs

```diff
@@ -0,0 +1,112 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+
+module Cardano.Ledger.Spec.STS.Update.Hupdate where
+
+import           GHC.Generics (Generic)
+--import           Data.Monoid.Generic (GenericMonoid (GenericMonoid),
+--                     GenericSemigroup (GenericSemigroup))
+import           Data.Map.Strict (Map)
+import qualified Data.Map.Strict as Map
+
+
+--import           Cardano.Crypto.Hash (Hash, HashAlgorithm)
+
+import           Control.State.Transition (Environment, PredicateFailure,
+                     STS, Signal, State, TRC (TRC), IRC (IRC), initialRules,
+                     judgmentContext, transitionRules)
+--import           Data.AbstractSize (HasTypeReps)
+
+import           Ledger.Core (Slot, BlockCount)
+import           Ledger.Core ((▷>=), (▷<=), (-.), (*.))
+
+
+-- import           Cardano.Ledger.Spec.STS.Update.Data ( SIPData
+--                                                      , Commit
+--                                                      )
+import qualified Cardano.Ledger.Spec.STS.Update.Data as Data
+
+
+
+-- | The Header Update STS
+-- Incorporates "update logic" processing
+-- at the block header level
+data HUPDATE hashAlgo
+
+data Env
+ = Env { k :: !BlockCount
+         -- ^ Chain stability parameter.
+       , currentSlot :: !Slot
```

Comment:
Do we need this the `currentSlot` here? Aren't we passing this as part of the signal?